Source/Particles/Collision/BinaryCollision/BinaryCollision.H

```diff
@@ -92,26 +93,43 @@ public:
             WARPX_ABORT_WITH_MESSAGE("Binary collision " + collision_name + " must have exactly two species.");
         }
 
+        const CollisionType collision_type = BinaryCollisionUtils::get_collision_type(collision_name, mypc);
+
         m_isSameSpecies = (m_species_names[0] == m_species_names[1]);
 
-        m_binary_collision_functor = CollisionFunctorType(collision_name, mypc, m_isSameSpecies);
+        m_binary_collision_functor = CollisionFunctor(collision_name, mypc, m_isSameSpecies);
 
         const amrex::ParmParse pp_collision_name(collision_name);
         pp_collision_name.queryarr("product_species", m_product_species);
+
+        // if DSMC the colliding species are also product species
+        if (collision_type == CollisionType::DSMC) {
+            m_product_species.push_back(m_species_names[0]);
+            if (!m_isSameSpecies) {
+                m_product_species.push_back(m_species_names[1]);
+            }
+            // rotate to have the colliding species first
+            std::rotate(
+                m_product_species.rbegin(),
+                m_product_species.rbegin() + 1 + (!m_isSameSpecies),
+                m_product_species.rend()
+            );
+        }
+
```

Comment:
```suggestion
        // if DSMC the colliding species are also product species
        // Therefore, we insert the colliding species at the beginning of `m_product_species`
        if (collision_type == CollisionType::DSMC) {
            m_product_species.insert( m_product_species.begin(), m_species_names.begin(), m_species_names.end() );
        }
```

Reply:
Does this work? It would make the code simpler, no?

Reply:
Thanks! That is much simpler!